src/integration/groovy/pl/allegro/tech/build/axion/release/SimpleIntegrationTest.groovy

```diff
@@ -77,6 +77,44 @@ class SimpleIntegrationTest extends BaseIntegrationTest {
         result.output.contains('2.0.0')
         result.task(":currentVersion").outcome == TaskOutcome.SUCCESS
     }
+		
+		def "should return tag with last version if useHighestVersion is set to false"() {
+				given:
+				buildFile('')
+	
+				runGradle('release', '-Prelease.version=1.0.0', '-Prelease.localOnly', '-Prelease.disableChecks')
+				runGradle('release', '-Prelease.version=1.5.0', '-Prelease.localOnly', '-Prelease.disableChecks')
+				
+				repository.commit(['*'], "commit after release-1.5.0")
+				
+				runGradle('release', '-Prelease.version=1.2.0', '-Prelease.localOnly', '-Prelease.disableChecks')
+	
+				when:
+				def result = runGradle('currentVersion')
+	
+				then:
+				result.output.contains('1.2.0')
+				result.task(":currentVersion").outcome == TaskOutcome.SUCCESS
+		}
+		
+		def "should return tag with highest version if useHighestVersion is set to true"() {
+				given:
+				buildFile('')
+	
+				runGradle('release', '-Prelease.version=1.0.0', '-Prelease.localOnly', '-Prelease.disableChecks')
+				runGradle('release', '-Prelease.version=1.5.0', '-Prelease.localOnly', '-Prelease.disableChecks')
+				
+				repository.commit(['*'], "commit after release-1.5.0")
+				
+				runGradle('release', '-Prelease.version=1.2.0', '-Prelease.localOnly', '-Prelease.disableChecks')
+	
+				when:
+				def result = runGradle('currentVersion', '-Prelease.useHighestVersion=true')
```

Comment:
no need for `=true`, this is a flag and it should be only checked using `hasProperty` disregarding the value